dialect/pgdialect/array.go

```diff
@@ -149,6 +149,8 @@ func (d *Dialect) arrayElemAppender(typ reflect.Type) schema.AppenderFunc {
 		if typ.Elem().Kind() == reflect.Uint8 {
 			return appendBytesElemValue
 		}
+	case reflect.Ptr:
+		return schema.PtrAppender(d.arrayElemAppender(typ.Elem()))
```

Comment:
A very concise implementation, but timestamp may need to be handled separately as they should also be considered a basic type.

Reply:
This should handle timestamps: https://github.com/uptrace/bun/pull/1111/commits/4c4e12aa7f27cf49189427da5104afb436af4348